### ISSUE TITLE
Request for VPC endpoint service

### ISSUE THREAD

**Issue by hanlins** [crossplane/provider-aws#956]
### What resource do you need?

We need to create [VPC endpoint service](https://docs.aws.amazon.com/vpc/latest/privatelink/endpoint-service.html).

### What is your use case?

We're SaaS vendor, and part of our job is to do LCM for AWS infra. We don't want to expose our service to the public internet, and we want to expose the service to a tenant securely via private link. It would be ideal if we have VPC endpoint service CRD so we can use k8s as a framework to manage the infra LCM.

### Would you be willing to contribute it using [code generator](https://github.com/crossplane/provider-aws/blob/master/CODE_GENERATION.md)?
I think so. My team has a fork of crossplane and we would be willing to contribute once we're done on our side. The main support we need is to get early feedback/insights from upstream and make decisions on the VPC endpoint service schema. Once that is done, we can consume our fork without worrying about the future divergence.

**Comment by haarchri** [comment]
We have Open PR https://github.com/crossplane/provider-aws/pull/817

**Comment by hanlins** [comment]
> We have Open PR #817

Hi @haarchri, thanks for the response! In the associated issue https://github.com/crossplane/provider-aws/issues/715#issuecomment-901564805, seems the PR is scoped only for VPC endpoint, and VPC endpoint service will NOT be included in the change. Do we need another PR dedicated for VPC endpoint service?

**Comment by haarchri** [comment]
yeah you are right - but its code-generated and it looks possible https://github.com/crossplane/provider-aws/pull/817/files#diff-b62878fa963633e85f0483e9506e5f777b7b0a5ec7a506fc3013ac836b3c3a3cL51 ;)

**Comment by hanlins** [comment]
Hey @haarchri, yup, it would be ideal if we can introduce both VPC endpoint and service support in https://github.com/crossplane/provider-aws/pull/817, but if people don't want to mix them together, we can add a followup PR after https://github.com/crossplane/provider-aws/pull/817 got merged.

**Comment by haarchri** [comment]
@hanlins i added https://github.com/crossplane/provider-aws/pull/1009 one open point i will check the next days is create & update tags
